Development/nmos/control_protocol_state.cpp

```diff
@@ -0,0 +1,361 @@
+                { U("NcEventId"), {make_nc_event_id_datatype()} },
+                { U("NcEventDescriptor"), {make_nc_event_descriptor_datatype()} },
+                { U("NcClassDescriptor"), {make_nc_class_descriptor_datatype()} },
+                { U("NcParameterConstraints"), {make_nc_parameter_constraints_datatype()} },
+                { U("NcDatatypeType"), {make_nc_datatype_type_datatype()} },
+                { U("NcDatatypeDescriptor"), {make_nc_datatype_descriptor_datatype()} },
+                { U("NcMethodResultClassDescriptor"), {make_nc_method_result_class_descriptor_datatype()} },
+                { U("NcMethodResultDatatypeDescriptor"), {make_nc_method_result_datatype_descriptor_datatype()} },
+                { U("NcMethodResultError"), {make_nc_method_result_error_datatype()} },
+                { U("NcDatatypeDescriptorEnum"), {make_nc_datatype_descriptor_enum_datatype()} },
+                { U("NcDatatypeDescriptorPrimitive"), {make_nc_datatype_descriptor_primitive_datatype()} },
+                { U("NcDatatypeDescriptorStruct"), {make_nc_datatype_descriptor_struct_datatype()} },
+                { U("NcDatatypeDescriptorTypeDef"), {make_nc_datatype_descriptor_type_def_datatype()} },
+                { U("NcEnumItemDescriptor"), {make_nc_enum_item_descriptor_datatype()} },
+                { U("NcFieldDescriptor"), {make_nc_field_descriptor_datatype()} },
+                { U("NcPropertyConstraintsNumber"), {make_nc_property_constraints_number_datatype()} },
+                { U("NcPropertyConstraintsString"), {make_nc_property_constraints_string_datatype()} },
+                { U("NcRegex"), {make_nc_regex_datatype()} },
+                { U("NcRolePath"), {make_nc_role_path_datatype()} },
+                { U("NcParameterConstraintsNumber"), {make_nc_parameter_constraints_number_datatype()} },
+                { U("NcParameterConstraintsString"), {make_nc_parameter_constraints_string_datatype()} },
+                { U("NcTimeInterval"), {make_nc_time_interval_datatype()} },
+                { U("NcTouchpointNmos"), {make_nc_touchpoint_nmos_datatype()} },
+                { U("NcTouchpointNmosChannelMapping"), {make_nc_touchpoint_nmos_channel_mapping_datatype()} },
+                { U("NcTouchpointResource"), {make_nc_touchpoint_resource_datatype()} },
+                { U("NcTouchpointResourceNmos"), {make_nc_touchpoint_resource_nmos_datatype()} },
+                { U("NcTouchpointResourceNmosChannelMapping"), {make_nc_touchpoint_resource_nmos_channel_mapping_datatype()} },
+                // Monitoring feature set
+                // See https://specs.amwa.tv/nmos-control-feature-sets/branches/main/monitoring/#datatypes
+                { U("NcConnectionStatus"), {make_nc_connection_status_datatype()} },
+                { U("NcPayloadStatus"), {make_nc_payload_status_datatype()} }
+            };
+        }
+
+        // insert control class, false if class already inserted
```

Comment:
```suggestion
        // insert control class descriptor, false if class already inserted
```

Reply:
This is along a theme of naming descriptors as descriptors so they don't get confused with class instances

Reply:
done